File: 2025-2/sesion-de-clase-06/SoftProg/SoftProgDBManager/src/main/java/pe/edu/pucp/inf30/softprog/db/DBManager.java
package pe.edu.pucp.inf30.softprog.db;

import java.sql.Connection;
import java.sql.SQLException;
import pe.edu.pucp.inf30.softprog.db.utils.TipoDB;
import pe.edu.pucp.inf30.softprog.db.utils.CadenaConexion;

public abstract class DBManager {
    protected String host;
    protected int puerto;
    protected String esquema;
    protected String usuario;
    protected String password;
    protected TipoDB tipoDB;
    
    protected DBManager() {}
    
    protected DBManager(String host, int puerto, String esquema, String usuario, 
                        String password, TipoDB tipoDB) {
        this.host = host;
        this.puerto = puerto;
        this.esquema = esquema;
        this.usuario = usuario;
        this.password = password;
        this.tipoDB = tipoDB;
    }
    
    public abstract Connection getConnection() throws SQLException, 
                                                      ClassNotFoundException;
    
    protected String cadenaConexion() {
        CadenaConexion cadena = new CadenaConexion.Builder()
                .tipoDB(this.tipoDB)
                .servidor(this.host)
                .puerto(this.puerto)
                .schema(this.esquema)
                .build();
        
        return cadena.toString();
    }
}

File: 2025-2/sesion-de-clase-21/dotnet/SoftProgDBManager/DBManager.cs
using PUCP.SoftProg.Db.Utils;
using System;
using System.Data.Common;

namespace PUCP.SoftProg.Db {
    public abstract class DBManager {
        protected string Servidor;
        protected int Puerto;
        protected string Esquema;
        protected string Usuario;
        protected string Password;
        protected TipoDB TipoDB;

        protected DBManager() { }

        protected DBManager(string servidor, int puerto, string esquema, string usuario,
                            string password, TipoDB tipoDB) {
            try {
                Servidor = servidor;
                Puerto = puerto;
                Esquema = esquema;
                Usuario = usuario;
                Password = Crypto.Decrypt(password);
                TipoDB = tipoDB;
            }
            catch (Exception ex) {
                Console.Error.WriteLine("Hubo un error al configurar la conexión a la base de datos: " 
                    + ex.Message);
                throw;
            }
        }

        public abstract DbConnection GetConnection();

        protected string CadenaConexion() {
            var conexion = new CadenaConexion.Builder()
                .ConServidor(Servidor)
                .ConPuerto(Puerto)
                .ConSchema(Esquema)
                .ConUsuario(Usuario)
                .ConPassword(Password)
                .ConTipoDB(TipoDB)
                .Build();

            return conexion.ToString();
        }
    }
}
